### PR TITLE
Feat #109: Farm-Endpoint und Error-Handling implementieren

### LINKED ISSUE #109: Farm-Endpoint
Als Entwicklerin möchte ich die in Issue #108 implementierte Kernlogik für Farmkäufe über das Netzwerk aufrufbar machen, damit das Frontend (die Spieler) Farmen über WebSockets kaufen kann und bei zu wenig Gold eine saubere, verarbeitbare Fehlermeldung erhält.

- [x] Neuer STOMP-Endpoint /app/buyFarm (bzw. später /app/rooms/{roomId}/buyFarm) der buyFarm aufruft
- [x] Neuer ErrorCode.INSUFFICIENT_GOLD ergänzt
- [x] player.farms wird im JSON-Broadcast serialisiert
- [x] Tests schreiben

DoD: 
- [x] ACs erfüllt · Tests grün 
- [x] PR-Review

### PR DESCRIPTION
**Pull request by SonjaBeisswaenger**
**Problem / Zweck**

Als Entwicklerin möchte ich die in Issue #108 implementierte Kernlogik für Farmkäufe über das Netzwerk aufrufbar machen, damit das Frontend Farmen über WebSockets kaufen kann und bei zu wenig Gold eine saubere, verarbeitbare Fehlermeldung erhält.

**Lösung**

* Neuer ErrorCode `INSUFFICIENT_GOLD` in der `ErrorCode` Enum-Klasse ergänzt.
* Neuer STOMP-Endpoint `@MessageMapping("/buyFarm")` im `WebSocketBrokerController` implementiert.
* Der Endpoint sucht den Spieler anhand der Session-ID, prüft dynamisch die Kosten und sendet bei Geldmangel eine gezielte Fehlermeldung (`/queue/errors`) an den auslösenden Spieler zurück.
* Bei erfolgreichem Kauf wird die `buyFarm`-Logik des `GameService` aufgerufen und der neue `GameState` via Broadcast (`/topic/game`) an alle Clients gesendet. Die Serialisierung von `player.farms` im JSON erfolgt automatisch durch die Data Class.
* Die Controller-Tests wurden um Erfolgs- und Fehlerfälle des neuen Endpoints sowie fehlende Edge-Cases (z.B. leere Session-IDs, Reconnects) erweitert. Die Branch-Coverage des Controllers liegt nun bei sicheren 87 %.

Closes #109

### CODE REVIEW

File: src/main/kotlin/at/aau/hexabrawl/websocketserver/controller/WebSocketBrokerController.kt [hunk 1]
```diff
@@ -87,6 +87,28 @@ class WebSocketBrokerController(
...
+        val state = gameService.getCurrentState()
+
+        // Spieler über die SessionId suchen
+        val player = state.players.find { it.sessionId == sessionId } ?: return null
+
+        // Kosten prüfen
+        val cost = GameService.FARM_BASE_COST + (player.farms * GameService.FARM_COST_INCREMENT)
+
+        if (player.gold < cost) {
+            // Deine eigene Hilfsfunktion nutzen!
```
Comment (STEFANLEGAT2002) [inline src/main/kotlin/at/aau/hexabrawl/websocketserver/controller/WebSocketBrokerController.kt]: bittte kommentar entfernen

File: src/main/kotlin/at/aau/hexabrawl/websocketserver/model/ErrorCode.kt [hunk 2]
```diff
@@ -4,5 +4,6 @@ enum class ErrorCode {
     NOT_YOUR_TURN,
     INVALID_MOVE,
     GAME_FULL,
-    GAME_NOT_STARTED
+    GAME_NOT_STARTED,
+    INSUFFICIENT_GOLD
```
Comment (STEFANLEGAT2002) [inline src/main/kotlin/at/aau/hexabrawl/websocketserver/model/ErrorCode.kt:8]: Neue Field. Passt

**Review by STEFANLEGAT2002 — CHANGES_REQUESTED** [review]

File: src/main/kotlin/at/aau/hexabrawl/websocketserver/controller/WebSocketBrokerController.kt (hunk 1 above)
Comment (SonjaBeisswaenger) [reply src/main/kotlin/at/aau/hexabrawl/websocketserver/controller/WebSocketBrokerController.kt]: Wird gemacht, danke.

**Comment by STEFANLEGAT2002** [comment]
Der neue Endpoint und das Error-Handling sind extrem sauber umgesetzt. Richtig stark, dass du auch an die Edge-Cases bei den Sessions gedacht hast. Danke
